pkg/oci/remote/options.go

```diff
@@ -78,6 +81,14 @@ func makeOptions(target name.Repository, opts ...Option) (*options, error) {
 	return o, nil
 }
 
+// WithSignaturePrefix is a functional option for overriding the default
+// signature tag prefix.
+func WithSignaturePrefix(prefix string) Option {
+	return func(o *options) {
+		o.SignaturePrefix = prefix
+	}
+}
+
```

Comment:
A lot of the plumbing seems to be adding the prefix support, which I don't fully understand the need for, if folks can customize the suffix.  Seems like a lot of knobs, where just suffix would hopefully suffice.

Reply:
@mattmoor I started with suffix only, until I realized that ECR Lifecycle Policies only allow for [tag prefixes](https://docs.aws.amazon.com/AmazonECR/latest/userguide/LifecyclePolicies.html#lp_tag_prefix_list).

I added the prefix option to account for that and left in the suffix option in case someone wanted to use something different than `.sig`. I'd be okay will pulling the suffix code out but really need the prefix code at the moment.

Reply:
`/me shakes fist at ECR`

I think I'd prefer we nail this for suffix (incl. stuff like `.att`) and then mirror everything for prefix.  Sorry if that's annoying.

Reply:
@mattmoor it sounds like this PR brought attention to a bigger code organization issue. I only exposed the existing `WithSignatureSuffix` to the CLI flags and added `WithPrefix` to the existing code base. Any chance we could merge this to get the feature in and then refactor how tags are formed in a separate PR?